lib/connection.js

```diff
@@ -112,15 +113,26 @@ class Connection extends EventEmitter {
   }
 
   // Send a handshake packet over the connection
-  //   appName - name of an application to connect to
+  //   app - string or object, application to connect to as 'name' or
+  //         'name@version' or { name, version }
   //   login - user name (optional)
   //   password - user password (optional)
   //   callback - callback function to invoke after the handshake is completed
   //
-  handshake(appName, login, password, callback) {
+  handshake(app, login, password, callback) {
+    let target;
+    if (typeof app === 'string') {
+      const [name, version] = app.split('@');
+      if (semver.valid(version)) target = [name, version];
+      else target = name;
```

Comment:
@aqrln as you can see in the comments below I'm waiting to get your approval to make all changes at once as I don't want to perform rebase multiple times for naught.